File: Inceptezpyspark-27/sparksql/Lab04_rdd_to_dataframe.py
from pyspark.sql import SparkSession


def main():
        
    spark = SparkSession.builder.appName("Lab04_rdd_to_dataframe").master("local").getOrCreate()
    sc = spark.sparkContext
    spark.sparkContext.setLogLevel("ERROR")
    
    #from pyspark import spark.implicits._
    
    rdd = sc.textFile("file:/home/hduser/hive/data/custs")
    
    rdd1 = rdd.map(lambda x : x.split(","))
    
    rdd2 = rdd1.filter(lambda x : len(x) == 5)
    
    rdd3 = rdd2.map(lambda x : (int(x[0]),x[1],x[2],int(x[3]),x[4]))
        
    df = rdd3.toDF(["custid","fname","lname","age","prof"])
     
    
    df.show()
    
    df.printSchema()
    
       
main()

File: InceptezSpark-27/src/inceptez/sparksql/Lab04_rdd_to_dataframe.scala
package inceptez.sparksql

import org.apache.spark.sql.SparkSession


object Lab04_rdd_to_dataframe {

def main(args:Array[String])=
{
  
  val spark = SparkSession.builder().appName("Lab04_rdd_to_dataframe").master("local").getOrCreate()
  val sc = spark.sparkContext
  sc.setLogLevel("ERROR")
  
  import spark.implicits._
  
  val rdd = sc.textFile("file:/home/hduser/hive/data/custs")
  
  val rdd1 = rdd.map(x => x.split(","))
  
  val rdd2 = rdd1.filter(x => x.length == 5)
  
  val rdd3 = rdd2.map(x => (x(0).toInt,x(1),x(2),x(3).toInt,x(4)))
  
  val df = rdd3.toDF("custid","fname","lname","age","prof")
  
  //or
  
  df.show()
  
  df.printSchema()
  
   
  val rdd31 = rdd2.map(x => customer(x(0).toInt,x(1),x(2),x(3).toInt,x(4)))
  //val rdd32 = rdd31.filter(x => x.age > 50 )
  
  val df1 = rdd31.toDF()
  
  
  df1.show()
  
  df1.printSchema()
  
}
    
  }

case class customer(custid:Int,fname:String,lname:String,age:Int,prof:String)

  
